### DISCUSSION TITLE
Custom model training Problem.

### DISCUSSION THREAD

**Discussion by SpiderWebEater** [openpifpaf/openpifpaf#496] [Q&A] [answered]
I am trying to run Openpipaf on my Jetson-Nano device to get a fast pose estimation. However, the pre-trained models available are large so they take a long time time to produce predictions and use a sizable portion of the machine's memory. To try and train a smaller network I use the process of knowledge distillation to train my small net using one of the pre-trained models (I tried as a teacher net shufflenetv2k30 and a teacher net shufflenetv2k16), on the dataset CocoKp, with MSE loss. attached I show Caf fields produced by my nets and the teacher nets(top plots are the results of the student and the bottom one the results of the teacher). As you can see from the images, the Caf fields of both nets are very similar, However, when I try to predict human poses using my student model the decoder returns an empty list. [images](https://www.dropbox.com/sh/oxbuyzahb2173n3/AABR4X_Jnm26UDHG1ydzalELa?dl=0)
I also tried training a student net using the command-line interface but changing the CompositeLoss with my loss function. For some reason, the nets whose Caf fields were closely resembling that of the teacher's started to give worse outputs. When I tried to retrain a pre-trained shufflenetv2k16 with the same configuration the net also got worse, but it still produced predictions. None of the nets I trained from scratch were able to produce predictions.
I am reaching out to you to ask a question about Openpifpaf's decoder. Shufflenetv2k16 and Shufflenetv2k30 produce CifCaf fields that are substantially different from one another, yet the decoder is able to produce Cif seeds with high confidence. My nets on the other hand are getting seeds with confidences of order 1e-9. My question is in regards to the low confidence of Cif seeds, I am not sure if I'm overlooking something, but I wanted to know how the decoder produces Cif seeds and Caf scores. i.e. does it work based on the outputs' relative structure or does it expect to get some specific outputs as answers to be able to produce reasonable seeds and scores?

**Comment by svenkreiss** [comment, marked as answer]
Hello @NjtehKourian , 

the code that produces seeds is here:
https://github.com/openpifpaf/openpifpaf/blob/9f6c2e63585430ebfddfd26aeecdb31227ebb6e0/openpifpaf/decoder/utils/cif_seeds.py#L30

So a Cif confidence must be above the given seed threshold (usually between 0.2 to 0.5) to produce seeds. Does that help?
